### DISCUSSION TITLE
Fallback to GitHub repo if `sourceUrl` not found for ghcr.io container image

### DISCUSSION THREAD

**Discussion by AverageMarcus** [renovatebot/renovate#23070] [Suggest an Idea]
### Tell us more.

I've got several images I pull from `ghcr.io` that don't have their source labels correctly set so the changelog isn't fetched. It'd be really useful if when working with a `ghcr.io` image that doesn't have a `sourceUrl` that renovate could default / fallback to using the associated GitHub repo to try and get the changelog details.

**Comment by viceice** [comment]
isn't easily doable, because the docker API doesn't provide that information

**Reply by AverageMarcus** [reply]
But the image URL is based on the GitHub URL. You just replace `ghcr.io` with `github.com`

**Reply by AverageMarcus** [reply]
You can even do a web request for the ghcr URL and it'll redirect to the packages page of the GitHub repo: e.g. Https://ghcr.io/tailscale/tailscale

**Reply by viceice** [reply]
> But the image URL is based on the GitHub URL. You just replace `ghcr.io` with `github.com`

wrong, only the org name matches, the image name can be arbitrary.

**Reply by viceice** [reply]
> You can even do a web request for the ghcr URL and it'll redirect to the packages page of the GitHub repo: e.g. Https://ghcr.io/tailscale/tailscale

we don't like to parse html pages, those change too frequently.

**Reply by AverageMarcus** [reply]
> wrong, only the org name matches, the image name can be arbitrary.

Oh really? I had no idea about that. I thought the image had to be associated with a repo. I've never seen any other example. 😞  That's frustrating then.

> we don't like to parse html pages, those change too frequently.

Yeah, I wasn't suggesting parsing HTML, just pointing out that the http redirects link up to a repo.

**Reply by viceice** [reply]
sample: https://github.com/orgs/visualon/packages

I've a lot images the which don't match a repo name.

**Comment by rarkins** [comment]
How can Renovate determine "the associated GitHub repo" if it only knows the image name?

**Reply by AverageMarcus** [reply]
The ghcr.io URL is related to the GitHub URL. You just replace `ghcr.io` with `github.com`.

**Reply by viceice** [reply]
> The ghcr.io URL is related to the GitHub URL. You just replace `ghcr.io` with `github.com`.

wrong

**Comment by AverageMarcus** [comment]
Yeah, looks like I misunderstood GHCR and wasn't aware that images could exist without an associated repo. I guess it would cause too many false positives to be useful. Thanks for your feedback.
